### PR TITLE
feature: isEqual interface method

### PR DESCRIPTION
**Pull request by EricRovell**
Implemented `.isEqual(color: Colord): boolean` utility method as discussed in #61.
Provided tests. Documentation is updated.

### CODE REVIEW

File: src/colord.ts [hunk 1]
```diff
@@ -174,6 +174,13 @@ export class Colord {
     if (typeof value === "number") return colord({ h: value, s: hsla.s, l: hsla.l, a: hsla.a });
     return round(hsla.h);
   }
+
+  /**
+   * Checks the color with another instance for equality.
+   */
+  public isEqual(color: Colord): boolean {
```
Comment (omgovich) [inline src/colord.ts:181]: It should accept any kind of input to make it easier to use (not only Colord instance).
Check `isReadable` method as an example.

```js
colord('white').isEqual('#FFF') // true
```

Comment (EricRovell) [reply src/colord.ts:181]: My bad, good point!

File: README.md [hunk 2]
```diff
@@ -518,6 +518,22 @@ colord({ r: 0, g: 0, v: 0 }).isValid(); // false
 
 </details>
 
+<details>
+  <summary><b><code>.isEqual()</code></b></summary>
```
Comment (omgovich) [inline README.md:522]: Please update the docs according to the latest changes:

```suggestion
  <summary><b><code>.isEqual(color2)</code></b></summary>
```

File: README.md [hunk 3]
```diff
@@ -518,6 +518,22 @@ colord({ r: 0, g: 0, v: 0 }).isValid(); // false
...
+Checks for color equality with another `Colord` instance.
+
+```js
+const color1 = colord("#ffffff")
+const color2 = colord("rgb(255, 255, 255)");
+const color3 = colord("rgb(255, 255, 255, 0.5)");
+
+color1.isEqual(color2) // true
+color1.isEqual(color3) // false
+```
```
Comment (omgovich) [inline README.md:533]: Please provide simpler examples like:

```js
colord('#000').isEqual('rgb(0, 0, 0)') // true
```

File: tests/colord.test.ts [hunk 4]
```diff
@@ -264,6 +264,20 @@ it("Rotates a hue circle", () => {
...
+    const instance = colord(lime[format] as AnyColor);
+    expect(instance.isEqual(colord(lime["hex"] as AnyColor))).toBe(true);
+    expect(instance.isEqual(colord(lime["rgb"] as AnyColor))).toBe(true);
+    expect(instance.isEqual(colord(lime["rgbString"] as AnyColor))).toBe(true);
+    expect(instance.isEqual(colord(lime["hsl"] as AnyColor))).toBe(true);
+    expect(instance.isEqual(colord(lime["hslString"] as AnyColor))).toBe(true);
+    expect(instance.isEqual(colord(lime["hsv"] as AnyColor))).toBe(true);
+    expect(instance.isEqual(otherInstance)).toBe(false);
+  }
+});
```
Comment (omgovich) [inline tests/colord.test.ts:279]: Please provide some examples that won't pass `Colord` instance to `isEqual`

Comment (EricRovell) [reply tests/colord.test.ts]: Honestly, forgot about that. Similar cases have been added with `AnyColor` types.

File: README.md (hunk 3 above)
Comment (EricRovell) [reply README.md]: Understood!

**Review by omgovich — APPROVED** [review]
Looks good! Thanks a lot! ❤️

**Comment by EricRovell** [comment]
Thank you, you are welcome :)

**Pull request merged**